### PR TITLE
Register with email

### PR DESCRIPTION
**Pull request by alexandernsalim**
Finish making CRUD API
Not done:
- Throw exception when email is found
- Throw exception when email not inputted

### CODE REVIEW

File: src/main/java/future/phase2/offlinetoonlinebazaar/OfflineToOnlineBazaarApplication.java [hunk 1]
```diff
@@ -13,8 +16,22 @@ public static void main(String[] args) {
 	}
 
 	@Bean
-	public ModelMapper modelMapper() {
-		return new ModelMapper();
+	public JavaMailSender getJavaMailSender(){
```
Comment (ekomanurung) [inline src/main/java/future/phase2/offlinetoonlinebazaar/OfflineToOnlineBazaarApplication.java:19]: ini pindahin ke class baru, misal namanya MailConfiguration. tambah anotasi @Configuration di class nya.

File: src/main/java/future/phase2/offlinetoonlinebazaar/service/CustomUserDetailsService.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        User user = userRepository.findByEmail(email);
+
+        if(user == null){
+            throw new UsernameNotFoundException("No user found with email: " + email);
+        }
+
+        boolean enabled = true;
+        boolean accountNonExpired = true;
+        boolean crendentialsNonExpired = true;
```
Comment (ekomanurung) [inline src/main/java/future/phase2/offlinetoonlinebazaar/service/CustomUserDetailsService.java:39]: langsung aja diset di constructor yg dibawah

File: src/main/java/future/phase2/offlinetoonlinebazaar/service/Impl/RoleServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+@Service
+public class RoleServiceImpl implements RoleService {
+
+    @Autowired
+    IdGenerator idGenerator;
+
+    @Autowired
+    RoleRepository roleRepository;
+
+    private MapperFactory mapperFactory = new DefaultMapperFactory.Builder().build();
```
Comment (ekomanurung) [inline src/main/java/future/phase2/offlinetoonlinebazaar/service/Impl/RoleServiceImpl.java:25]: ini dipindah ke class baru ya, jadiin Component, biar bisa di reuse di tmpat lain. jd ga declare berkali kali. contoh:

public class BeanMapper {

  private static MapperFactory mapperFactory = new DefaultMapperFactory.Builder().build();
  private static MapperFacade mapper = mapperFactory.getMapperFacade();

/**
   *
   * @param source object yg mau diconvert
   * @param clazz tipe class destination yg mau di convert
   * @param <S>
   * @param <C>
   * @return list of object
   * digunakan untuk mapping Object -> AnotherObject
   */
  public static <S, C> C map(S source, Class<C> clazz) {
    if (source == null) {
      return null;
    }
    return mapper.map(source, clazz);
  }

  /**
   *
   * @param source object yg mau diconvert
   * @param clazz tipe class destination yg mau di convert
   * @param <S>
   * @param <C>
   * @return list of object
   * digunakan untuk mapping Collection<Object> -> Collection<AnotherObject>
   */
  public static <S, C> List<C> mapAsList(Iterable<S> source, Class<C> clazz) {
    if (source == null) {
      return null;
    }
    return mapper.mapAsList(source, clazz);
  }
}

File: src/main/java/future/phase2/offlinetoonlinebazaar/OfflineToOnlineBazaarApplication.java (hunk 1 above)
Comment (ekomanurung) [reply src/main/java/future/phase2/offlinetoonlinebazaar/OfflineToOnlineBazaarApplication.java:19]: @Configuration
public class MailConfiguration {

  @Bean
  public JavaMailSender getJavaMailSender(){
    JavaMailSenderImpl mailSender = new JavaMailSenderImpl();

    mailSender.setHost("smtp.gmail.com");
    mailSender.setPort(587);
    mailSender.setUsername("offlinetoonline.bazaar@gmail.com");
    mailSender.setPassword("OffToOnAdmin");

    Properties props = mailSender.getJavaMailProperties();

    props.put("mail.transport.protocol", "smtp");
    props.put("mail.smtp.auth", "true");
    props.put("mail.smtp.starttls.enable", "true");
    props.put("mail.debug", "true");

    return mailSender;
  }
}

File: src/main/java/future/phase2/offlinetoonlinebazaar/service/CustomUserDetailsService.java (hunk 2 above)
Comment (alexandernsalim) [reply src/main/java/future/phase2/offlinetoonlinebazaar/service/CustomUserDetailsService.java:39]: Maksudnya constructor yg mn bg?

File: src/main/java/future/phase2/offlinetoonlinebazaar/service/Impl/RoleServiceImpl.java (hunk 3 above)
Comment (alexandernsalim) [reply src/main/java/future/phase2/offlinetoonlinebazaar/service/Impl/RoleServiceImpl.java:25]: Kalau private gimana cara makainya ntar bg?

Comment (ekomanurung) [reply src/main/java/future/phase2/offlinetoonlinebazaar/service/Impl/RoleServiceImpl.java:25]: tinggal manggil nya by class. itu kan static, jadi 
Beanmapper.map(s,d)

File: src/main/java/future/phase2/offlinetoonlinebazaar/service/CustomUserDetailsService.java (hunk 2 above)
Comment (ekomanurung) [reply src/main/java/future/phase2/offlinetoonlinebazaar/service/CustomUserDetailsService.java:39]: return new org.springframework.security.core.userdetails.User(
                user.getEmail(),
                user.getPassword(),
                true,   // gak perlu assignment yg diatas
                accountNonExpired,
                crendentialsNonExpired,
                accountNonLocked,
                getAuthorities(user.getRoles())
        );

File: src/main/java/future/phase2/offlinetoonlinebazaar/service/Impl/RoleServiceImpl.java (hunk 3 above)
Comment (alexandernsalim) [reply src/main/java/future/phase2/offlinetoonlinebazaar/service/Impl/RoleServiceImpl.java:25]: fungsinya sama seperti yg contoh abg buat?

**Pull request merged**